crates/task-impls/src/vid.rs

```diff
@@ -100,12 +108,13 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 // send the block to the VID dispersal function
                 self.event_stream
                     .publish(HotShotEvent::BlockReady(
-                        VidDisperse {
+                        VidDisperse::from_membership(
                             view_number,
-                            payload_commitment: vid_disperse.commit,
-                            shares: vid_disperse.shares,
-                            common: vid_disperse.common,
-                        },
+                            vid_disperse.commit,
+                            vid_disperse.shares,
+                            vid_disperse.common,
```

Comment:
You could make `from_membership` just take the entire `vid_disperse` as a single arg so as to reduce the number of args. But it's not worth blocking this merge for it.

Reply:
Good idea, will implement